lib/util/cache/repository/impl/client/local/index.ts

```diff
@@ -0,0 +1,36 @@
+import upath from 'upath';
+import { GlobalConfig } from '../../../../../../config/global';
+import { logger } from '../../../../../../logger';
+import { outputCacheFile, readCacheFile } from '../../../../../fs';
+import type { CacheClient, RepoCacheRecord } from '../../../types';
+
+export class LocalRepoCache implements CacheClient {
+  constructor(private platform: string, private repository: string) {}
+
```

Comment:
This local cache client contains the extracted fs handling logic from `lib/util/cache/repository/impl/local.ts` now renamed 
`repository/impl/repository-cache-impl.ts`

Reply:
move that to the cache base class and remove this class. we don't need so many layers.

in this pr we now have
- index
- RepositoryCacheImpl
- CacheClient

so i prefer that the init.ts calls the Factory and calls index.ts/setCache with result.
That way we can completle remove `RepositoryCacheImpl`, funtionallity should be moved to `RepoCacheBase` and all clients should extend that.